docs/sphinx/sections/api/apidocs/libraries/dagster-pipes.rst

```diff
@@ -22,8 +22,15 @@ Context
 
 ----
 
+Advanced
+--------
+
+The APIs in the following sections are primarily used by those who need more control over Pipes than the simplified objects, such as ``dagster.PipesSubprocessClient``, can offer. Most Pipes users won't need to use these APIs in their Pipes implementation.
```

Comment:
I would say

"Most Pipes users won't need to use these APIs unless they are customizing the Pipes protocol"